docs/docs/user/references/components.md

```diff
@@ -0,0 +1,245 @@
+- Submit schemas to the registry if provided (optional)
+
+#### destroy
+
+Refer to [KubernetesApp](#kubernetesapp).
+
+#### reset
+
+Not implemented.
+
+#### clean
+
+Not implemented.
+
+## StreamsApp
+
+Sub class of [_KafkaApp_](#kafkaapp).
+
+### Usage
+
+Configures a
+[streams-bootstrap app](https://github.com/bakdata/streams-bootstrap) which in
+turn configures a
+[Kafka Streams app](https://kafka.apache.org/documentation/streams/).
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/streams-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+Refer to [KafkaApp](#kafkaapp).
+
+#### destroy
+
+Refer to [KafkaApp](#kafkaapp).
+
+#### reset
+
+Runs a reset job in the cluster that is deleted by default after a succesful
+completion.
+
+- Reset the consumer group offsets
+- Reset Kafka Streams state
+
+#### clean
+
+- Reset Kafka Streams state
+- Delete the app's output topics
+- Delete all associated schemas in the Schema Registry
+- Delete the consumer group
+
+## Producer
+
+Sub class of [_KafkaApp_](#kafkaapp).
+
+### Usage
+
+Publishes (writes) events to a Kafka cluster.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/producer.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+Refer to [KafkaApp](#kafkaapp).
+
+#### destroy
+
+Refer to [KafkaApp](#kafkaapp).
+
+#### reset
+
+Not implemented, producers are stateless.
+
+#### clean
+
+- Delete the output topics of the Kafka producer
+- Delete all associated schemas in the Schema Registry
+
+## KafkaSinkConnector
+
+Sub class of [KafkaConnector](/docs/user/references/config.md)
+
+### Usage
+
+Lets other systems pull data from Apache Kafka.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kafka-sink-connector.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+- Deploy the Kafka connector
+- Create the output topics if provided (optional)
+- Register schemas in the Schema registry if provided (optional)
+
+#### destroy
+
+The associated Kafka connector is removed.
+
+#### reset
+
+Reset the consumer group offsets.
+
+#### clean
+
+- Delete associated consumer group
+- Delete configured error topics
+
+## KafkaSourceConnector
+
+Sub class of [KafkaConnector](/docs/user/references/config.md)
+
+### Usage
+
+Lets other systems push data to Apache Kafka.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kafka-source-connector.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+- Deploy the Kafka connector
+- Create the output topics if provided (optional)
+- Register schemas in the Schema registry if provided (optional)
+
+#### destroy
+
+The associated Kafka connector is removed.
+
+#### reset
+
+Delete state associated with the connector
```

Comment:
Maybe provide a link or an explanation of what you mean by `state`. I don't know if this that obvious or I am missing here something

Reply:
tbh I don't know what it means either, but since most of the colleagues used it I assumed it must be trivial. I should've asked..